src/js/components/modals/JobStopRunModal.js

```diff
@@ -35,8 +36,15 @@ class JobStopRunModal extends mixin(StoreMixin) {
   handleButtonConfirm() {
     const { selectedItems, jobID } = this.props;
     // TODO DCOS-8763 introduce support for multiple job run IDs
+
     if (selectedItems.length === 1) {
-      MetronomeStore.stopJobRun(jobID, selectedItems[0]);
+      const jobRunID = selectedItems[0];
+
+      if (jobID == null || jobRunID == null) {
+        return;
+      }
+
+      MetronomeClient.stopJobRun(jobID, jobRunID).subscribe();
```

Comment:
Do we need a take(1) or something to make sure it finishes and no subscription is tangling in the air?

Reply:
i had a similar concern although i believe i only moved code on a structural level.  
the observable from http-service's `request` `complete`s once the request is finished. so we don't need to make doubly sure here.
if there's some edge case that would still requires us to jump through hoops here i think that should be fixed on a library-level to not convolute the semantics of a "request" (as i understand them 😊)

Reply:
👍